pkgs/servers/mail/system-sendmail/default.nix

```diff
@@ -0,0 +1,16 @@
+{ writeScriptBin }:
+
+writeScriptBin "sendmail" ''
+  #!/bin/sh
+
+  if which sendmail > /dev/null 2>&1; then
+    exec sendmail "$@"
+  elif [ -x /run/wrappers/bin/sendmail ]; then
+    exec /run/wrappers/bin/sendmail "$@"
+  elif [ -x /run/current-system/sw/bin/sendmail ]; then
+    exec /run/current-system/sw/bin/sendmail "$@"
+  else
+    echo "Unable to find system sendmail." >&2
+    exit 1
+  fi
+''
```

Comment:
This is very ugly and I'd rather not have something like this in nixpkgs, but I don't know of a better solution right now.

Reply:
I agree, but already had to introduce it for `mdadm` in https://github.com/NixOS/nixpkgs/pull/38067 (OpenSMTPD's `sendmail` doesn't require setuid like postfix's), and putting it in a separate package will allow to share code with it (see https://github.com/NixOS/nixpkgs/pull/49229 that I'll rebase once this gets finished, and probably others I don't already know about because I haven't tested them on OpenSMTPD yet and they support only postfix because they have the security wrapper path hard-coded).

Thank you for your review! I'll review your other points when getting back to my computer where I can test things :)